### PR TITLE
Use the new i18n-loader-webpack-plugin

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

* webpack-config: Use `@automattic/i18n-loader-webpack-plugin`
    
  Provide the new plugin as part of the standard set of plugins.

* jetpack: Use `@automattic/i18n-loader-webpack-plugin`
    
  Configure the new plugin and remove the hack Instant Search was using to
  work around the problem the plugin was created to fix.

#### Jetpack product discussion

PT: p9dueE-3MG-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Is CI happy?
* Configure your site for a non-English locale, activate Instant Search, then search. Is the text in Instant Search's popup translated?
  * If you're using the Jetpack Docker env, you might do `jetpack docker exec /var/scripts/fakepot.sh /var/www/html/wp-content/plugins/jetpack` to generate `en_piglatin` and `en_rtl` locales.
    ss
* Do the above on wpcom too.
  * Currently to test it on a wpcom sandbox, you'll also have to go into `wp-content/mu-plugins/jetpack-packages/production/` and `composer update automattic/jetpack-assets=1.14.x-dev` to get the new version of the assets package.
  * Also add this filter hook somewhere:
    ```php
    function wpcom_jetpack_i18n_state( $data ) {
        $data['baseUrl'] = 'https://widgets.wp.com/languages/';
        return $data;
    }
    add_filter( 'jetpack_i18n_state', 'wpcom_jetpack_i18n_state' );
    ```
    (that'll have to be added to D70962-code before we merge it)

### CODE REVIEW

File: projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php
Comment (kangzj) [inline projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php:157]: in WPCOM, the translation path was resolved to `mu-plugins` if `/plugins` not specified here.

**Review by kangzj — DISMISSED** [review]
Hi @anomiex wonderful job! So excited we could get this resolved so soon.
This works for me and looks good. We might want to test WPCOM before merging, left an inline comment.

File: projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php
Comment (anomiex) [reply projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php:157]: It should be ok since the path now isn't being determined by wpcom PHP code, but testing is a good idea.

Looks like we may have to wait until #21932 is merged first so the TC build on D70962-code will succeed. Or merge this into #21932 then test it with a D70957-code that contains the changes from here too, if reviewers would rather.

Comment (anomiex) [reply projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php:157]: So it turns out testing this is kind of a pain, because the Fusion patch doesn't include the changes to the Assets package but those are required for it to work.

Once I manually apply those in my sandbox, it *almost* works. It tries to fetch the right path, but apparently wpcom blocks access to `.json` files inside `wp-content/languages/`.

Comment (anomiex) [reply projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php:157]: Figured it out, and updated the testing instructions so it should be easy for someone else to test it too.

**Review by kraftbj — DISMISSED** [review]
This looks good and tests well on Jetpack. Let's do it.

**Comment by anomiex** [comment]
Needs re-approval after merge conflicts.

**Comment by anomiex** [comment]
Sigh. So the wpcom stuff isn't happy because D71392-code isn't merged yet, but it can't be due to p3topS-X5-p2

**Comment by anomiex** [comment]
Hacked around the wpcom problem, this should be ready for review again.

**Review by sdixon194 — DISMISSED** [review]

**Review by kraftbj — APPROVED** [review]

**Comment by anomiex** [comment]
r236678-wpcom

**Pull request merged**
